### PR TITLE
Combine participation and release forms

### PR DESCRIPTION
**Pull request by awisema**
Draft until we have confirmation that they want to deploy.

### CODE REVIEW

**Comment by awisema** [comment]
Combines Participation and ROI Lines, and adds hourglass for items in need of renewal:
image
Collect both Participation and ROI dates on form:
image
Move download blank button and offer both patient and verbal form options:
image

File: app/models/health/agency_performance.rb [hunk 1]
```diff
@@ -143,10 +142,13 @@ def patient_referrals
     # end
 
     def consent_dates
+      # Participation signature was moved into ReleaseForm
       @consent_dates ||= Health::Patient.
-        has_signed_participation_form.
-        # where(hpf_t[:signature_on].between(@range)).
-        pluck(:patient_id, hpf_t[:signature_on].to_sql).to_h
+        joins(:release_forms).
+        pluck(:patient_id, hrf_t[:participation_signature_on].to_sql).to_h
```
Comment (eanders) [inline app/models/health/agency_performance.rb]: I _think_ you'll want an order by on here.  Also, I think you can `where(hrf_t[:participation_signature_on].not_eq(nil))`.  The `to_h` on the end will be problematic without an order clause if a patient has more than one release form.

File: app/models/health/engagement_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      [:valid, 'Participation Form', edit_client_health_participation_form_path(@client, signed_participation_form), true, nil]
+    end
+
+    def release_form_status
+      signed_release_form = @patient.release_forms.recent.first
+      return [:no_signed_form, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, nil] if signed_release_form.blank?
+      return [:too_old, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, "Last signed on #{signed_release_form.signature_on}"] if signed_release_form.signature_on < @valid_after
+      return [:expired, 'Participation and Release of Information', new_client_health_release_form_path(@client), true, "Expired on #{signed_release_form.signature_on + 2.years}"] if signed_release_form.signature_on + 2.years <= Date.current
+
+      [:valid, 'Participation and Release of Information', edit_client_health_release_form_path(@client, signed_release_form), true, nil]
```
Comment (eanders) [inline app/models/health/engagement_calculator.rb]: I'm probably missing it, but doesn't this need to check the participation signature somewhere in there?

File: app/models/health/health_file.rb [hunk 3]
```diff
@@ -28,19 +28,21 @@ def file_not_too_large
...
+      errors.add :file, 'File must be a PDF' if content_type != 'application/pdf'
     end
 
     def title
       self.class.model_name.human
     end
 
     def signature
-      try(:participation_form).try(:signature_on) || try(:release_form).try(:signature_on)
+      try(:release_form).try(:participation_signature_on) || try(:release_form).try(:signature_on)
```
Comment (eanders) [inline app/models/health/health_file.rb]: this is checking any signature, not fully signed.  Is that correct?

File: app/models/health/release_form.rb [hunk 4]
```diff
@@ -29,6 +29,7 @@ class ReleaseForm < HealthBase
     include HealthFiles
 
     validates :signature_on, presence: true
+    validates :participation_signature_on, presence: true
```
Comment (eanders) [inline app/models/health/release_form.rb:32]: so both signatures are required, always?  I guess that might clear up some of my previous notes.

**Review by eanders — APPROVED** [review]

File: app/models/health/engagement_calculator.rb (hunk 2 above)
Comment (awisema) [reply app/models/health/engagement_calculator.rb]: As you noticed below, they can't sign a release without having a participation signature, so this shouldn't be a problem.

File: app/models/health/agency_performance.rb (hunk 1 above)
Comment (awisema) [reply app/models/health/agency_performance.rb]: I'll replace this with a group -- they only need to sign the participation agreement once, so we can just use their earliest one.

File: app/models/health/health_file.rb (hunk 3 above)
Comment (awisema) [reply app/models/health/health_file.rb]: This is a strange bit of STI behavior, refactored.

**Pull request merged**